util/profile/profile.go

```diff
@@ -3,13 +3,28 @@ package profile
 import (
 	"net/http"
 	"net/http/pprof"
+	"os"
+
+	"github.com/argoproj/argo-cd/v2/util/env"
 )
 
+var enableProfilerFilePath = env.StringFromEnv("ARGOCD_ENABLE_PROFILER_FILE_PATH", "/home/argocd/params/profiler.enabled")
+
+func wrapHandler(handler http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if data, err := os.ReadFile(enableProfilerFilePath); err == nil && string(data) == "true" {
+			handler.ServeHTTP(w, r)
+		} else {
+			http.Error(w, "Enabled profile endpoint in 'argocd-cmd-params-cm' ConfigMap", http.StatusUnauthorized)
```

Comment:
```suggestion
			http.Error(w, "Profiler endpoint is not enabled in 'argocd-cmd-params-cm' ConfigMap", http.StatusUnauthorized)
```

it's _not_ enabled in this case

Reply:
ops, fixed